vochain/app.go

```diff
@@ -629,7 +630,16 @@ func (app *BaseApplication) DeliverTx(req abcitypes.RequestDeliverTx) abcitypes.
 func (app *BaseApplication) Commit() abcitypes.ResponseCommit {
 	data, err := app.State.Save()
 	if err != nil {
-		log.Fatalf("cannot save state: %s", err)
+		log.Fatalf("cannot save state: %v", err)
+	}
+	if app.Height()%50000 == 0 && !app.IsSynchronizing() {
+		startTime := time.Now()
+		log.Infof("performing a state snapshot on block %d", app.Height())
+		if _, err := app.State.snapshot(); err != nil {
+			log.Fatalf("cannot make state snapshot: %v", err)
```

Comment:
If it fails the execution should stop since there is no point for failing (only if the disk is full or there are some serious I/O issues). Thus should not fail.